1059 netcdf hyrax ondemand update

hyi opened the pull request:
This resolves on-demand netCDF resource update feed to Hyrax server when private netCDF resources are made public so that links of data services provided by Hyrax server on resource landing page are instantaneously valid and available once the netCDF resource is made public. The cron job that runs the update script every midnight will still be scheduled to clean up synlinked netCDF resources as needed for cases when a public netCDF resource is made private or deleted. I have tested it out in my local development environment and made sure it all works as expected. @pkdash @gantian127 Could you give a quick code review and see if you can give a +1?

hydrobuild commented:
Can one of the admins verify this patch?

gantian127 commented:
+1

hyi commented:
@pkdash Can you review also once you have a chance? I will merge this once you give a +1 also.

pkdash commented on hs_core/views/utils.py:
```diff
@@ -59,6 +61,26 @@ def upload_from_irods(username, password, host, port, zone, irods_fnames, res_fi
         res_files.append(UploadedFile(file=tmpFile, name=fname, size=size))
 
+# run the update script on hyrax server via ssh session for netCDF resources on demand
+# when private netCDF resources are made public so that all links of data services
```
@hyi I am assuming this function is executed every time the resource ```public``` attribute is reset (either True or False). Is that right?

hyi replied:
@pkdash Not exactly, this function is only executed every time when the netcdf resource ```public``` attribute is set to True, i.e., when a private resource is made public. Since the script will be run as a cron job on hyrax server to clean things up when public resource is made private or deleted, there is no need to run it on-demand for those cases. Your question made me realize I did not ensure this in the code logic. I will make changes tomorrow morning and let you know once I am done so that you can give a final review. Thanks for catching this.

hyi commented:
@pkdash Have addressed your review comment to call the script execution only when a private netCDF resource is made public as it is intended. Please review again and see if you can +1 now.

pkdash commented:
@hyi I have never used paramiko module, so know nothing about it. Assuming you have tested the on-demand update script to work, here is my +1.

hyi commented:
@pkdash Yes, I have test it out in my local dev env. Once it is deployed, @gantian127 will test it to make sure it works. I will merge it then.
